sophia/src/term/literal/_convert.rs

```diff
@@ -0,0 +1,495 @@
+//! Public exported by parent-module `literal`.
+//!
+
+use super::*;
+use crate::ns::xsd;
+use crate::term::{Iri, Result, StaticTerm, TermData, TermError};
+use crate::triple::stream::{SinkError, SourceError, StreamResult};
+
+/// Provide an IRI to specify the semantic datatype.
+pub trait DataType {
+    /// Static datatype of this Rust-type.
+    fn dt() -> &'static Iri<&'static str>;
+
+    /// Datatype of `self`.
+    ///
+    /// This can be used for `enum`s where each variant may has its own
+    /// datatype.
+    fn dt_of(&self) -> &'static Iri<&'static str> {
+        Self::dt()
+    }
+
+    fn dt_term() -> StaticTerm {
+        Iri::from(Self::dt()).into()
+    }
+
+    fn dt_term_of(&self) -> StaticTerm {
+        Iri::from(Self::dt_of(self)).into()
+    }
+}
+
+/// Representation as an RDF literal.
+pub trait AsLiteral<TD: TermData>: DataType {
+    /// Create an RDF literal, representing `self`.
+    fn as_literal(&self) -> Literal<TD>;
+
+    /// Blanked implementation to directly get a term.
+    fn as_term(&self) -> Term<TD> {
+        self.as_literal().into()
+    }
+}
+
+/// Create `self` from an RDF literal.
+pub trait FromLiteral: Sized {
+    /// Error raised if literal text is malformed.
+    type Error: std::error::Error;
+
+    /// Direct from a literal.
+    fn from_literal<TD: TermData>(lit: &Literal<TD>) -> Result<Self, Self::Error>;
+
+    /// Get a term that is actually a literal.
+    ///
+    /// # Error
+    ///
+    /// The blanked implementation throws `SourceError(UnexpectedKindOfTerm)`
```

Comment:
```suggestion
    /// The default implementation throws `SourceError(UnexpectedKindOfTerm)`
```
I think you wanted to write "blanket", but what you really meant is "default" :)